contracts/eos/BancorConverter/BancorConverter.cpp

```diff
@@ -51,92 +53,64 @@ ACTION BancorConverter::create(name owner, symbol_code token_code, double initia
         st.multi_token, "transfer"_n,
         make_tuple(get_self(), owner, initial_supply_asset, string("setup"))
     ).send();
-}
-
-ACTION BancorConverter::setmaxfee(uint64_t maxfee) {
-    require_auth(get_self());
-
-    settings settings_table(get_self(), get_self().value);
-    const auto& st = settings_table.get("settings"_n.value, "set token contract first");
 
-    string error = string("max fee must be lower or equal to ") + std::to_string(MAX_FEE);
-    check(maxfee <= MAX_FEE, error.c_str());
-    check(maxfee != st.max_fee, "setting same value as before");
-    settings_table.modify(st, same_payer, [&](auto& s) {
-        s.max_fee = maxfee;
-    });
+    // MIGRATE DATA to V2
+    migrate_converters_v2( token_code );
 }
 
-ACTION BancorConverter::setstaking(name staking) {
-    check(is_account(staking), "invalid staking account");
-    require_auth(get_self());
+[[eosio::action]]
+void BancorConverter::setparams( const settings_params params )
+{
+    require_auth( get_self() );
+    BancorConverter::settings _settings( get_self(), get_self().value );
+    const auto settings = _settings.get_or_default();
```

Comment:
nice, wasn't aware of that one

Reply:
Yea pretty handy one indeed.

`singleton` tables are pretty easy to use for things such as settings.

- get()
- get_or_create()
- get_or_default()
- set()
- remove()

https://github.com/EOSIO/eosio.cdt/blob/master/libraries/eosiolib/contracts/eosio/singleton.hpp